engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2256,7 +2143,9 @@ - (BOOL)hasText {
 
 - (void)insertText:(NSString*)text {
   if (self.temporarilyDeletedComposedCharacter.length > 0 && text.length == 1 && !text.UTF8String &&
-      [text characterAtIndex:0] == [self.temporarilyDeletedComposedCharacter characterAtIndex:0]) {
+      [text characterAtIndex:0] ==
+          [self.temporarilyDeletedComposedCharacter
+              characterAtIndex:self.temporarilyDeletedComposedCharacter.length - 1]) {
```

Comment:
is this intended change (0 -> length-1)?

Reply:
Yes, I revisited https://github.com/flutter/flutter/issues/111494 and I think this is the right fix. See https://github.com/flutter/flutter/issues/111494#issuecomment-2981917269

Reply:
Also I tested this on iOS 16 and confirmed it fixed the crash as far as I can tell.